### PR TITLE
add workflow diagrams

### PR DESCRIPTION
**Pull request by kornel-filep**
## What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
Adds docs on the planned deployment workflows

## What is the current behavior? (You can also link to an open issue here)
-
## What is the new behavior (if this is a feature change)?
-
## Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
-
## Other information:
-

### CODE REVIEW

**Comment by szikszail** [comment]
@kornel-filep please add the drawio files as well so that later they can be edited if needed

**Review by szikszail — CHANGES_REQUESTED** [review]
~~@kornel-filep please add the drawio files as well so that later they can be edited if needed~~

**Review by szikszail — CHANGES_REQUESTED** [review]
Please ignore the previous comment, I just found it.
Could you please create a markdown file describing the CI/CD workflow?

**Comment by kornel-filep** [comment]
> Please ignore the previous comment, I just found it. Could you please create a markdown file describing the CI/CD workflow?

I have added a description to the diagrams. Please let me know if I missed anything, or if I should write down in more detail.

**Review by szikszail — DISMISSED** [review]

File: .github/docs/branch-deploy.drawio.png
Comment (taughty) [inline .github/docs/branch-deploy.drawio.png:1]: Artifacts seem to be inconsistent, i.e. `build` job archives _application build artifacts_ and `deploy` job downloads _application artifact_. Also, in the *Archived artifacts* box it is referred to as _site_.

File: .github/docs/main-deploy.drawio.png
Comment (taughty) [inline .github/docs/main-deploy.drawio.png:1]: Artifacts seem to be inconsistent, i.e. `build` job uploads _Pages artifacts_ and `deploy` job downloads _application artifact_.

File: .github/docs/stack-deploy.drawio.png
Comment (taughty) [inline .github/docs/stack-deploy.drawio.png:1]: Is this pipeline going to use the _CDK Output JSON_ archieved by the `stack-deploy` pipeline? In this case you might want to add it in a box under the _GitHub Environment variables_ the same way as on the `stack-deploy` diagram.

**Review by taughty — COMMENTED** [review]
You might want to fix some incorrect casing in source files, i.e.:
- _Github Actions_ -> _GitHub Actions_
- _Github pages_ -> _GitHub Pages_
- _Cloudfront_ -> _CloudFront_

File: .github/docs/stack-deploy.drawio.png
Comment (kornel-filep) [reply .github/docs/stack-deploy.drawio.png:1]: No, this pipeline should not use the output JSON file.
We archive it just as an output of the CDK execution, the necessary stuff is saved to new variables for the environment, like `AWS_BUCKET_NAME`.

**Comment by kornel-filep** [comment]
@taughty addressed your comments with a new commit. Please take a look and resolve if suitable.

**Review by szikszail — APPROVED** [review]

**Review by taughty — APPROVED** [review]

**Pull request merged**